zip/zip.go

```diff
@@ -0,0 +1,73 @@
+package zip
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/dsnet/compress/bzip2"
+)
+
+const archiveExtension = ".tar.bz2"
+
+func CreateArchiveFromDir(source string) error {
+	return createArchive(source, func(tarWriter *tar.Writer) error {
+		return filepath.Walk(source, func(file string, fi os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			return writeToArchive(tarWriter, file, fi)
+		})
+	})
+}
+
+func CreateArchiveFromFile(name string) error {
+	fi, err := os.Stat(name)
+	if err != nil {
+		return err
+	}
+	return createArchive(fi.Name(), func(tarWriter *tar.Writer) error {
+		return writeToArchive(tarWriter, name, fi)
+	})
+}
+
+func writeToArchive(tarWriter *tar.Writer, fileName string, fileInfo os.FileInfo) error {
+	if !fileInfo.Mode().IsRegular() {
+		return nil
+	}
+	header, err := tar.FileInfoHeader(fileInfo, fileName)
+	if err != nil {
+		return err
+	}
+	header.Name = filepath.ToSlash(fileName)
+	if err := tarWriter.WriteHeader(header); err != nil {
+		return err
+	}
+	file, err := os.Open(fileName)
```

Comment:
This is already being passed in as fileInfo

Reply:
But then I'd need to open the actual file for copying its contents into the tar writer.
I am not sure I understand the problem...

Reply:
Weird, I don't know if I had an older version come up when I commented here or what.  It may have displayed the older commit, but I swear there was an os.Open above.